### ISSUE TITLE
mac build broken

### ISSUE THREAD

**Issue by doutriaux1** [UV-CDAT/uvcdat#991]
system python merge broke build on macs ( @aashish24 please do not do that again :smile: )
PYTHONPATH isn't set correctly then pip/easy_install complains about PYTHONPATH not containing directory where we try to install packages.

**Comment by jbeezley** [comment]
See #988, I think that takes care of it.

**Comment by aashish24** [comment]
@doutriaux1 we had system python branch sitting there for weeks (we had to do it :cry: ). Hopefully the fix pushed by @jbeezley will take care of that issue. Also, note that it seems to be broken beforehand (looking at the changes).

**Comment by ThomasMaxwell** [comment]
I just merged my 2.1.1 contributions branch into master and tried building under OSX 10.8.5.   I encountered the following build error.   Charles, is this the error you were referring to in #991?   Is there a fix?

TEST FAILED:
  /Developer/Projects/EclipseWorkspace/uvcdat/master/build1/install/lib/python2.7/site-packages/
  does NOT support .pth files
  error: bad install directory or PYTHONPATH

  You are attempting to install a package to a directory that is not on PYTHONPATH and which Python does not read ".pth" files from.  The  installation directory you specified (via --install-dir, --prefix, or  the distutils default setting) was:

      /Developer/Projects/EclipseWorkspace/uvcdat/master/build1/install/lib/python2.7/site-packages/

  and your PYTHONPATH environment variable currently contains:

 '/Developer/Projects/EclipseWorkspace/uvcdat/master/build1/install/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages'

**Comment by remram44** [comment]
I'm also seeing this build failure. [Same log as Travis](https://travis-ci.org/UV-CDAT/uvcdat/jobs/48775517#L4294):
```
  You are attempting to install a package to a directory that is not
  on PYTHONPATH and which Python does not read ".pth" files from.  The
  installation directory you specified (via --install-dir, --prefix, or
  the distutils default setting) was:

      /Users/remirampin/Documents/programming/uvcdat/builds/4f89f6eb442faf03979a4e0706fc5fd4d24aa352/install/lib/python2.7/site-packages/

  and your PYTHONPATH environment variable currently contains:

      '/Users/remirampin/Documents/programming/uvcdat/builds/4f89f6eb442faf03979a4e0706fc5fd4d24aa352/install/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages'
```

**Comment by jbeezley** [comment]
That is the error that #988 addresses.

**Comment by ThomasMaxwell** [comment]
I merged the fix-pythonpath-apple branch into my branch and rebuilt.   That seemed to fix the problem.

**Comment by doutriaux1** [comment]
building on my mac, if it works for me, will merge

**Comment by ThomasMaxwell** [comment]
After building with the fix-pythonpath-apple branch merged I got an error on uvcdat startup complaining:

No such file or directory: '…/install/share/vcs/wmo_symbols.json'

There does not seem to be a ../install/share/vcs directory.

From: <Maxwell>, "Maxwell, Thomas P. (GSFC-606.2)[SCIENCE APPLICATIONS INTL CORP]" <thomas.maxwell@nasa.gov<mailto:thomas.maxwell@nasa.gov>>
Date: Thursday, January 29, 2015 12:48 PM
To: UV-CDAT/uvcdat <reply@reply.github.com<mailto:reply@reply.github.com>>
Subject: Re: [uvcdat] mac build broken (#991)

I merged the fix-pythonpath-apple branch into my branch and rebuilt.   That seemed to fix the problem.

**Comment by doutriaux1** [comment]
I see it too. That means that now because we changed the PYTHONAPTH thing it gets installed somewhere else, will fix right away.

**Comment by jbeezley** [comment]
The path it is using comes from here[1].  For one, that function should turn `UVCDAT_SETUP_PATH` into an absolute path.  I suspect that generating the return value at configure time from the cmake variable `PYTHON_SITE_PACKAGES_PREFIX` may fix it, but I don't know what other ramifications that may have.

[1] https://github.com/UV-CDAT/uvcdat/blob/master/CMake/cdat_modules_extra/install.py.in#L267

**Comment by doutriaux1** [comment]
ok it is fixed and now works for me. Can someone confirm it works for them as well and I will merge.

**Comment by remram44** [comment]
It works on Travis. I also did `make IPYTHON` here without problem.

**Comment by doutriaux1** [comment]
ok merging then!

**Comment by remram44** [comment]
New error:
```
Scanning dependencies of target CDAT
[ 69%] Creating directories for 'CDAT'
[ 70%] No download step for 'CDAT'
[ 70%] No patch step for 'CDAT'
[ 70%] No update step for 'CDAT'
[ 71%] No configure step for 'CDAT'
[ 71%] No build step for 'CDAT'
[ 71%] Performing install step for 'CDAT'
Running cdat install with path: /Users/remirampin/bin:/opt/local/bin:/opt/local/sbin:/opt/local/Library/Frameworks/Python.framework/Versions/Current/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/MacGPG2/bin
Configuring & installing scripts.
Reading configuration: installation/pp.py
Traceback (most recent call last):
  File "/Users/remirampin/Documents/programming/uvcdat/uvcdat/CMake/install.py", line 10, in <module>
    execfile(install_script_path)
  File "/Users/remirampin/Documents/programming/uvcdat/uvcdat/CMake/../installation/install.py", line 950, in <module>
Your Python checked OK!
Running: /Users/remirampin/Documents/programming/uvcdat/uvcdat/CMake/../installation/contrib.py
Version is:  2.1.0-122-gcd3f51e
    main(arglist)
  File "/Users/remirampin/Documents/programming/uvcdat/uvcdat/CMake/../installation/install.py", line 740, in main
    configure(configuration_files)
  File "/Users/remirampin/Documents/programming/uvcdat/uvcdat/CMake/../installation/install.py", line 437, in configure
    repr(os.environ['EXTERNALS']),
TypeError: %d format: a number is required, not str
```

**Comment by remram44** [comment]
The error is in [this commit](https://github.com/UV-CDAT/uvcdat/commit/cd3f51ee5713b8af623d4cdceea7ad5effed15e0#commitcomment-9506723).

**Comment by ThomasMaxwell** [comment]
Using the current master branch I get the following error when building on OSX 10.8.5:

[ 87%] Performing install step for 'CDAT'
Running cdat install with path: /Developer/Projects/EclipseWorkspace/uvcdat/master/build1/install/bin:/Developer/Projects/EclipseWorkspace/uvcdat/master/build1/install/Externals/bin:/Developer/Projects/EclipseWorkspace/uvcdat/master/build1/install/Library/Frameworks/Python.framework/Versions/2.7/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/opt/X11/bin:/usr/local/git/bin:/opt/local/bin:/opt/local/sbin:/Developer/Applications/Xcode.app/Contents/MacOS
Configuring & installing scripts.
Reading configuration: installation/pp.py
Traceback (most recent call last):
  File "/Developer/Projects/EclipseWorkspace/uvcdat/master/source/CMake/install.py", line 10, in <module>
    execfile(install_script_path)
  File "/Developer/Projects/EclipseWorkspace/uvcdat/master/source/CMake/../installation/install.py", line 950, in <module>
Your Python checked OK!
Running: /Developer/Projects/EclipseWorkspace/uvcdat/master/source/CMake/../installation/contrib.py
Version is:  2.1.0-192-g2834f0e
    main(arglist)
  File "/Developer/Projects/EclipseWorkspace/uvcdat/master/source/CMake/../installation/install.py", line 740, in main
    configure(configuration_files)
  File "/Developer/Projects/EclipseWorkspace/uvcdat/master/source/CMake/../installation/install.py", line 437, in configure
    repr(os.environ['EXTERNALS']),
ValueError: unsupported format character '(' (0x28) at index 369
Make Errors detected:


CMake Error at /Developer/Projects/EclipseWorkspace/uvcdat/master/build1/CMake/cdat_python_install_step.cmake:27 (message):
  Error in Python Install.  1


make[2]: *** [CDAT-prefix/src/CDAT-stamp/CDAT-install] Error 1
make[1]: *** [CMakeFiles/CDAT.dir/all] Error 2
make: *** [all] Error 2

**Comment by jbeezley** [comment]
To save other people looking into it, I pushed #998 preliminarily.  I'll update if the PR if it fixes the build.

**Comment by jbeezley** [comment]
#998 works for me and it passes on travis.  Please give that a try.

**Comment by ThomasMaxwell** [comment]
Merging the branch escape-install-pt fixes this issue for me.

**Comment by remram44** [comment]
Is this fixed now? #988 #995 #998 have been merged in.

**Comment by durack1** [comment]
@doutriaux1 wanna add the bug label (for @chaosphere2112's readme: #913)
